lua/telescope-fzf-native/download_library.lua

```diff
@@ -0,0 +1,122 @@
+local uv = vim.loop
+local plugin_path = string.sub(debug.getinfo(1).source, 2, #"/lua/telescope-fzf-native/download_library.lua" * -1)
+local releases_url = "https://github.com/nvim-telescope/telescope-fzf-native.nvim/releases/download"
+
+local get_platform = function()
+    if vim.fn.has("win32") == 1 then
+        return 'windows'
+    end
+
+    if vim.fn.has("mac") then
+        return 'macos'
+    end
+
+    return "ubuntu"
+end
+
+local get_valid_compiler = function(platform)
+    if platform == 'windows' then
+        return 'cc'
+    elseif platform == 'macos' then
+        return 'gcc'
+    elseif platform == 'linux' then
+        return 'gcc'
+    end
+end
+
+local path_join = function(strings)
+    local path_separator = (platform == "windows") and '\\' or '/'
+
+    return table.concat(strings, path_separator)
+end
+
+local write_async = function(path, txt, flag)
+    uv.fs_open(path, flag, 438, function(open_err, fd)
+        assert(not open_err, open_err)
+        uv.fs_write(fd, txt, -1, function(write_err)
+            assert(not write_err, write_err)
+            uv.fs_close(fd, function(close_err)
+                assert(not close_err, close_err)
+            end)
+        end)
+    end)
+end
+
+local spawn = function(cmd, on_exit)
+    local stdout = uv.new_pipe()
+
+    local buffer = ""
+
+    local real_cmd = table.remove(cmd, 1)
+
+    uv.spawn(real_cmd, {
+        args = cmd,
+        stdio = { nil, stdout },
+        verbatim = true
+    }, function()
+        stdout:read_stop()
+        if (type(on_exit) == "function") then
+            on_exit(buffer)
+        end
+    end)
+
+    uv.read_start(stdout, function(err, data)
+        assert(not err, err)
+        if data then
+            buffer = buffer .. data
+        end
+    end)
+
+end
+
+local download = function(options)
```

Comment:
lets just immediately return the function rather than having an additional local